src/main/scala/io/iohk/ethereum/proof/MptProofVerifier.scala

```diff
@@ -0,0 +1,52 @@
+package io.iohk.ethereum.proof
+
+import akka.util.ByteString
+import cats.syntax.either._
+import cats.syntax.functor._
+import io.iohk.ethereum.db.dataSource.EphemDataSource
+import io.iohk.ethereum.db.storage.{NodeStorage, SerializingMptStorage, StateStorage}
+import io.iohk.ethereum.jsonrpc.{MptProofError, KeyNotFoundInRebuidMpt, UnableRebuildMpt}
+import io.iohk.ethereum.mpt.{ByteArrayEncoder, ByteArraySerializable, MerklePatriciaTrie, MptNode}
+
+object MptProofVerifier {
```

Comment:
is this a 'test' only mechanism? or should it be available for 'prod' code also?

Reply:
Moved to tests.